hail/python/hail/context.py

```diff
@@ -311,7 +313,10 @@ def init(sc=None,
         List of regions to run jobs in when using the Batch backend. Use :data:`.ANY_REGION` to specify any region is allowed
         or use `None` to use the underlying default regions from the hailctl environment configuration. For example, use
         `hailctl config set batch/regions region1,region2` to set the default regions to use.
-
+    storage_location_allow_list:
+        A list of cloud storage locations that the ``hailtop.batch.backend.ServiceBackend`` should be permitted to
+        read from or write to, even if their default policy is to use "cold" storage. Should look like
+        ``{"cloud1": ["location1", "location2"]}``. The only supported cloud name is ``"gcp"``.
```

Comment:
the user isn't specifying full URIs in the config variable, just bucket names, which is why they need to pass in the cloud name as well

Reply:
I feel like it is common though to see those listed as `gs://my-bucket` and `https://my-account.blob.core.windows.net/my-container`, or if you want just the account, `https://my-account.blob.core.windows.net`.

Reply:
We need the account *and* container to get hotness, I think. Iris, is that right?

IIRC, Iris asked about how to handle multiple clouds and we went with this nested structure to make clear that the buckets / accounts&containers were not paths but accounts/containers. The Azure backend is also rather wordy.

Reply:
for azure, we just need the account name, since there's no per-container setting.

i don't have a strong opinion on having the config variable be like `gs://my-bucket,https://my-account.blob.core.windows.net` vs `gcs:my-bucket,azure:my-account`; for the google bucket, it seems like there's no real difference in effort between the two; for the azure account, the latter seems less annoying to type out, but then again, the former could be copy-pasted from the web ui, so i guess it really just depends whether the user is more likely to have the storage account open in their browser while they're doing this, or whether they just know the name of the storage account already

Reply:
Fair point about ease of use. In the end it doesn't matter too much, especially if we can include a couple concrete examples in the docs?

Reply:
sounds good, added example uses to the config reference page and the help text for the `hailctl` variable